### PR TITLE
Add QuFPN per #580.

### PR DESCRIPTION
**Pull request by mingxingtan**
Also fix learning rate, add epoch summary, and add other code clean up

### CODE REVIEW

**Comment by mingxingtan** [comment]
@shivam-raj  With this PR, I am able to train D0 with fpn_name=qufpn, and improve validation accuracy from 34.0 to 35.0.  Could you please help review this PR? Thanks!

File: efficientdet/keras/fpn_configs.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+
+def qufpn_config(min_level, max_level, weight_method=None, quad_method=None):
+  """A dynamic quad fpn config that can adapt to different min/max levels."""
+  # It extends the idea of BiFPN, and has four paths:
+  #   (up_down -> bottom_up) + (bottom_up -> up_down).
+  # See test for an example for level 2 and 7.
+  p = hparams_config.Config()
+  p.weight_method = weight_method or 'fastattn'
+  p.quad_method = quad_method or p.weight_method
```
Comment (shivam-raj) [inline efficientdet/keras/fpn_configs.py:83]: @mingxingtan 
p.quad_method shouldn't be ever set to unweighted sum (if p.weight_method is set to 'sum') , it produces bad results owing to equal contribution from the two parallel paths.
Actually you were right earlier, we don't really require two weight methods. p.quad_method is always set to 'fastattn' regardless of p.weight_method

```suggestion
  p.quad_method = 'fastattn'
```

**Comment by shivam-raj** [comment]
@mingxingtan That's an even more impressive result !! 
Did you try training the bigger models while increasining the fpn_cell_repeats by 2 each time we scale the backbone?

**Comment by mingxingtan** [comment]
> @mingxingtan That's an even more impressive result !!
> Did you try training the bigger models while increasining the fpn_cell_repeats by 2 each time we scale the backbone?

@shivam-raj  I will try to train a big model later when I get the resources.
I am submitingt this PR now to unblock other PRs.
